### DISCUSSION TITLE
Cannot throw error with custom properties

### DISCUSSION THREAD

**Discussion by TasinIshmam** [apollographql/apollo-server#5345] [Q&A] [answered]
Hi, according to the [Error handling page](https://www.apollographql.com/docs/apollo-server/data/errors/#including-custom-error-details) in the Apollo Server docs, it's possible to add arbitrary fields to the error's ```extensions``` object. The documentation also gives an example of a ```userInputError``` that throws some additional properties.

However, looking at the [source](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-errors/src/index.ts) of the apollo-server-error package, I see that only ```UserInputError``` accepts an optional properties argument (for logging additional properties). All the other error subclasses, such as ```AuthenticationError```, ```SyntaxError```, etc. do not have this. 

Code that can be used to validate this: 

```javascript
import { AuthenticationError, UserInputError} from "apollo-server-errors";

// this works fine
throw new UserInputError('Invalid argument value', {
          argumentName: 'id'
        });

// typescript compiler raises error (TS2554: Expected 1 arguments, but got 2.)
throw new AuthenticationError('Invalid authentication', {
          argumentName: 'id'
        });

```

Is this intentional behavior or a bug? Thought I should mention it here first before issuing a bug report just in case I'm misinterpreting the docs somehow. 

Thanks

**Comment by glasser** [comment, marked as answer]
See https://github.com/apollographql/apollo-server/issues/5307 / https://github.com/apollographql/apollo-server/pull/5309
This is being improved in the imminent 3.0.0. Some other errors haven't had that added yet; you can look into the concerns @bristoljon ran into with the other error classes if you'd like.

**Reply by TasinIshmam** [reply]
Good to know, thanks.
